### ISSUE TITLE
Unable to update challenge :: ownership issues?

### ISSUE THREAD

**Issue by AndyZaa** [acmesh-official/acme.sh#2660]
Hi Neil,

I used your acme.sh process to install SSL on six Wordpress sites hosted at GoDaddy using Deluxe Linux Hosting with cPanel. I installed all six in October 2018 and they have auto-renewed beautifully every two months since then. This week I received an email from the Let's Encrypt Expiry Bot that two of the domains did not renew automatically and had less than 20 days to expiration.

acme.sh is set for auto-update and is running the most recent version. I spent quite a few hours today trying to track down the problem with the renewals. I can force renew 4 of the sites no problem using "acme.sh --renew -d example.com --force". The other 2 cannot update the challenge. I read and tried various techniques from other solutions but no joy. I'll attach two debug outputs - one is from a failed renew, and the second is from a successful renew. Any direction from you and your team would be much appreciated.

I originally installed all 6 certs using:
acme.sh --issue -d example.network -d www.example.network -w /home/username/public_html/example
acme.sh  --deploy  -d example.network -d www.example.network  --deploy-hook cpanel_uapi

Today I tried to re-issue one of the failed ones using the DNS API guide for GoDaddy at https://github.com/Neilpang/acme.sh/wiki/dnsapi
export GD_Key="xxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
export GD_Secret="xxxxxxxxxxxxxxxx"
acme.sh --issue --dns dns_gd -d example.com -d www.example.com
No joy there either.

Of the two failed renews, both are web hosted at GoDaddy, one is DNS hosted at GoDaddy, one is DNS hosted at Namecheap.

By the way, the interface here wouldn't accept my TXT files so I renamed them as LOG and then it worked.

Thank you and Happy New Year!
Andy

[acme.sh debug failed renew.log](https://github.com/Neilpang/acme.sh/files/4014755/acme.sh.debug.failed.renew.log)
[acme.sh debug success renew.log](https://github.com/Neilpang/acme.sh/files/4014756/acme.sh.debug.success.renew.log)

**Comment by AndyZaa** [comment]
Hi Neil,
1. Just verifying that I have acme.sh v2.8.4 installed which I believe is the most recent.
2. I didn't see .well-known/acme-challenge in my webroot so I created them with 777 privileges. 
Still no joy. The two sites that will not renew properly run out on January 20th so I'm in a bit of a time crunch. Is there another route I should try? I don't know what else to try and I'm running out of time.
Thanks,
Andy

**Comment by Neilpang** [comment]
the failed log uses webroot mode.
the success log uses  dns mode.

The problem is that your url at:   http://youdomain.com/.well-known/acme-challenge/test.txt  can not be accessed.

Maybe your htaccess file is changed.

**Comment by AndyZaa** [comment]
Hi Neil,

 

I’m sorry to be thick about this. I just can’t figure out where I’ve gone wrong.

 

I checked the htaccess and it has not changed. Today I tried the following:

 

1. deleted acme.sh folder

 

2. reinstalled acme.sh

 

3. reissue cert

acme.sh --issue -d fuglydog.com -d www.fuglydog.com -w /home/fweew92k6plb/public_html/fuglydog

which failed, then with 

acme.sh --issue -d fuglydog.com -d www.fuglydog.com -w /home/fweew92k6plb/public_html/fuglydog --debug 2

The debug file is attached.

 

4. Last week I made .well-known/acme-challenge/ directories in my public_html directory and installed a test file

https://zaayenga.net/.well-known/acme-challenge/TestFile.txt

is accessible and able to be read.

 

After the above operations 1-3 there is a file in

https://zaayenga.net/fuglydog/.well-known/acme-challenge/uLERPlCKifqI4CgSmb7kkmvAN8940JWBxAEOOyp-Dps

that is accessible and able to be read.

zaayenga.net is my GoDaddy base directory.

 

Looking at the error messages, the response failures are associated with

https://fuglydog.com/fuglydog/.well-known/acme-challenge/FShtDPfIzFhihmHrkuHa_eKMSlEECljt7jqPqgJrn84

but fuglydog.com/fuglydog/ doesn't exist, it would have to be 

zaayenga.net/fuglydog/ or the internal directory home/fweew92k6plb/public_html/fuglydog

 

Can you point me in the right direction?

 

Thank you very much,

Andy Z

**Comment by AndyZaa** [comment]
Hi Neil,

I still have four of my six sites renewing certs reliably and properly, and two of them continue to fail. The two failed sites are way out of date now so they throw up the insecure site errors. All six sites were installed using acme.sh at the same time, and worked perfectly renewing the certs for over a year. Suddenly at the end of last year I received messages that the two failed sites were in danger of expiration, and that's where I started having all these problems.

Your last message was that:
"the failed log uses webroot mode.
the success log uses dns mode.
The problem is that your url at: http://youdomain.com/.well-known/acme-challenge/test.txt can not be accessed.
Maybe your htaccess file is changed."

Fuglydog.com is one of the failing cert web sites.
I did check that https://fuglydog.com/.well-known/acme-challenge/test.txt can be accessed.
I looked at the htaccess file and it has not changed.

I don't understand why the failed sites use webroot mode and the successful sites use DNS mode. Did I do something wrong with the acme.sh commands? When I deleted the acme.sh folder and reinstalled I used this format:
acme.sh --issue -d fuglydog.com -d www.fuglydog.com -w /home/fweew92k6plb/public_html/fuglydog

I've also tried issuing the certs using the GoDaddy API as in your doc. After successfully adding the text records the system just goes into 10 second loops forever checking for _acme-challenge.fuglydog.com and _acme-challenge.www.fuglydog.com but always finding them not valid yet:
fweew92k6plb@a2plcpnl0871 [~]$ acme.sh --issue --dns dns_gd -d fuglydog.com -d www.fuglydog.com
[Wed Feb 26 18:16:34 MST 2020] Multi domain='DNS:fuglydog.com,DNS:www.fuglydog.com'
[Wed Feb 26 18:16:34 MST 2020] Getting domain auth token for each domain
[Wed Feb 26 18:16:37 MST 2020] Getting webroot for domain='fuglydog.com'
[Wed Feb 26 18:16:37 MST 2020] Getting webroot for domain='www.fuglydog.com'
[Wed Feb 26 18:16:37 MST 2020] Adding txt value: pdmOL09fCsKZ8WOJDFpQriQ3NZD8XUSX7jaLQdEPwNA for domain:  _acme-challenge.fuglydog.com
[Wed Feb 26 18:16:38 MST 2020] Adding record
[Wed Feb 26 18:16:38 MST 2020] Added, sleeping 10 seconds
[Wed Feb 26 18:16:49 MST 2020] The txt record is added: Success.
[Wed Feb 26 18:16:49 MST 2020] Adding txt value: Ptm_LJYp0CPpydyaXbHplmz9kDtx1rdS2uCUHC898P4 for domain:  _acme-challenge.www.fuglydog.com
[Wed Feb 26 18:16:50 MST 2020] Adding record
[Wed Feb 26 18:16:51 MST 2020] Added, sleeping 10 seconds
[Wed Feb 26 18:17:02 MST 2020] The txt record is added: Success.
[Wed Feb 26 18:17:02 MST 2020] Let's check each dns records now. Sleep 20 seconds first.
[Wed Feb 26 18:17:23 MST 2020] Checking fuglydog.com for _acme-challenge.fuglydog.com
[Wed Feb 26 18:17:23 MST 2020] Not valid yet, let's wait 10 seconds and check next one.
[Wed Feb 26 18:17:35 MST 2020] Checking www.fuglydog.com for _acme-challenge.www.fuglydog.com
[Wed Feb 26 18:17:35 MST 2020] Not valid yet, let's wait 10 seconds and check next one.
[Wed Feb 26 18:17:47 MST 2020] Let's wait 10 seconds and check again.
...and looping

Should I completely remove the failed sites certs and try reinstalling from a clean environment? If so, what elements should I find and delete? If possible I would like to leave the four successful sites alone so I don't end up with six failures.

With a GoDaddy site, is the GoDaddy.com domain API the best method to issue certs?

Acme.sh is at the current rev of v2.8.6

Thanks,
Andy

**Comment by Neilpang** [comment]
your file `https://fuglydog.com/.well-known/acme-challenge/test.txt` is accessable now.

please use webroot mode to issue a cert again.

**Comment by AndyZaa** [comment]
Thank you, Neil. I spent hours today trying to install the broken certs. I have another one of the six sites that I received a Let's Encrypt certificate expiration notice so that one is not auto-renewing either. That makes three of the six failing after more than a year of error free renewals.

I tried many combinations of the webroot mode html directory for the one site, fuglydog.com, that is failing to renew or install. That site is not so critical so it's a good test site. I'm running in test mode so I don't lock up my LetsEncrypt account with too many requests.

The basic format I used is:
acme.sh --test --issue -d fuglydog.com -d www.fuglydog.com -w /home/fweew92k6plb/public_html
with changes to the html directory in attempts to get a cert installed.
Some of the variations I tried:
/home/fweew92k6plb/public_html - from your directions, fails with "Verify error:Invalid response from https://fuglydog.com/fuglydog"
/fweew92k6plb/public_html
/home/fuglydog
/home/fweew92k6plb/public_html/fuglydog
...and so on
All variations failed, either with "mkdir: cannot create directory : Permission denied; or with "Verify error:Invalid response from https://fuglydog.com/fuglydog/.well-known/acme-challenge/****; or with "Verify error:Invalid response from https://fuglydog.com/fuglydog"

Finally I tried a test on my zaayenga.net site which is the primary site.
I ran: acme.sh --test --issue -d zaayenga.net -d www.zaayenga.net -w /home/fweew92k6plb/public_html
and it worked fine!

I apologize for not mentioning my type of Godaddy account. I have a Deluxe Linux Hosting with cPanel account. There is one primary site, zaayenga.net, at /home/fweew92k6plb/public_html. The other sites are under that one. For instance, the fuglydog.com site is at /home/fweew92k6plb/public_html/fuglydog. Is that structure perhaps the problem? Is it possible that an update to acme.sh broke the renew and install functions of such a structure? 

Please let me know if I should run a test with debug on for further troubleshooting. Thank you so much for a wonderful tool!

- Andy

**Comment by AndyZaa** [comment]
Hi Neil,

In the interest of providing as much troubleshooting detail as possible, I tried issuing and renewing the sucessful cert in my primary root, zaayenga.net, as well as two of the failing certs that are under it, fuglydog.com and rozanosigns.com.

When I initially installed and deployed the certs in 2018, I used these formats which all worked just fine and then renewed every 60 days without error:
acme.sh --issue -d zaayenga.net -d www.zaayenga.net -w /home/fweew92k6plb/public_html
acme.sh --issue -d fuglydog.com -d www.fuglydog.com -w /home/fweew92k6plb/public_html/fuglydog
acme.sh --issue -d rozanosigns.com -d www.rozanosigns.com -w /home/fweew92k6plb/public_html/rozanosigns

As you'll see below, now the primary site installs and renews fine, but the sites under that do not and come up with a weird verification path which does not exist: "fuglydog.com/fuglydog/". No matter what combination of parameters I try for the sub-directories, I can't get it to use a proper .well-known directory path. 

Where should the .well-known directory be? There is one in the main root (/home/fweew92k6plb/public_html). The .acme.sh directory where the certs are being generated and stored is there as well. The failing sites certs are being generated and are there as well in the .acm.sh directory.

By the way, I don't presume that the problem is coming from your script. It's very possible that GoDaddy has changed something that is breaking acme.sh.

Does this help at all? Is there any other information I can provide?

Thanks,
Andy

-----------------------------------------------

acme.sh --test --force --issue -d zaayenga.net -d www.zaayenga.net -w /home/fweew92k6plb/public_html
[Tue Mar  3 11:39:02 MST 2020] Using stage ACME_DIRECTORY: https://acme-staging-v02.api.letsencrypt.org/directory
[Tue Mar  3 11:39:03 MST 2020] Multi domain='DNS:zaayenga.net,DNS:www.zaayenga.net'
[Tue Mar  3 11:39:03 MST 2020] Getting domain auth token for each domain
[Tue Mar  3 11:39:05 MST 2020] Getting webroot for domain='zaayenga.net'
[Tue Mar  3 11:39:05 MST 2020] Getting webroot for domain='www.zaayenga.net'
[Tue Mar  3 11:39:05 MST 2020] zaayenga.net is already verified, skip http-01.
[Tue Mar  3 11:39:05 MST 2020] www.zaayenga.net is already verified, skip http-01.
[Tue Mar  3 11:39:05 MST 2020] Verify finished, start to sign.
[Tue Mar  3 11:39:05 MST 2020] Lets finalize the order, Le_OrderFinalize: https://acme-staging-v02.api.letsencrypt.org/acme/finalize/12636650/77747184
[Tue Mar  3 11:39:07 MST 2020] Download cert, Le_LinkCert: https://acme-staging-v02.api.letsencrypt.org/acme/cert/facb6c85d60b5ea355b09fdcdb2316f95785
[Tue Mar  3 11:39:07 MST 2020] Cert success.
-----BEGIN CERTIFICATE-----
xxxxxxxxxxxxxxxxxxxxxx
-----END CERTIFICATE-----
[Tue Mar  3 11:39:07 MST 2020] Your cert is in  /home/fweew92k6plb/.acme.sh/zaayenga.net/zaayenga.net.cer
[Tue Mar  3 11:39:07 MST 2020] Your cert key is in  /home/fweew92k6plb/.acme.sh/zaayenga.net/zaayenga.net.key
[Tue Mar  3 11:39:07 MST 2020] The intermediate CA cert is in  /home/fweew92k6plb/.acme.sh/zaayenga.net/ca.cer
[Tue Mar  3 11:39:07 MST 2020] And the full chain certs is there:  /home/fweew92k6plb/.acme.sh/zaayenga.net/fullchain.cer

acme.sh --renew --force -d zaayenga.net -d www.zaayenga.net
[Tue Mar  3 11:14:58 MST 2020] Renew: 'zaayenga.net'
[Tue Mar  3 11:14:58 MST 2020] Multi domain='DNS:zaayenga.net,DNS:www.zaayenga.net'
[Tue Mar  3 11:14:58 MST 2020] Getting domain auth token for each domain
[Tue Mar  3 11:15:01 MST 2020] Getting webroot for domain='zaayenga.net'
[Tue Mar  3 11:15:01 MST 2020] Getting webroot for domain='www.zaayenga.net'
[Tue Mar  3 11:15:01 MST 2020] zaayenga.net is already verified, skip http-01.
[Tue Mar  3 11:15:01 MST 2020] www.zaayenga.net is already verified, skip http-01.
[Tue Mar  3 11:15:01 MST 2020] Verify finished, start to sign.
[Tue Mar  3 11:15:01 MST 2020] Lets finalize the order, Le_OrderFinalize: https://acme-staging-v02.api.letsencrypt.org/acme/finalize/12636650/77744412
[Tue Mar  3 11:15:03 MST 2020] Download cert, Le_LinkCert: https://acme-staging-v02.api.letsencrypt.org/acme/cert/fabea8623e9ad2cef83ce16918097f6874c9
[Tue Mar  3 11:15:03 MST 2020] Cert success.
-----BEGIN CERTIFICATE-----
xxxxxxxxxxxxxxxxxxxxxx
-----END CERTIFICATE-----
[Tue Mar  3 11:15:03 MST 2020] Your cert is in  /home/fweew92k6plb/.acme.sh/zaayenga.net/zaayenga.net.cer
[Tue Mar  3 11:15:03 MST 2020] Your cert key is in  /home/fweew92k6plb/.acme.sh/zaayenga.net/zaayenga.net.key
[Tue Mar  3 11:15:03 MST 2020] The intermediate CA cert is in  /home/fweew92k6plb/.acme.sh/zaayenga.net/ca.cer
[Tue Mar  3 11:15:03 MST 2020] And the full chain certs is there:  /home/fweew92k6plb/.acme.sh/zaayenga.net/fullchain.cer

acme.sh --test --issue -d fuglydog.com -d www.fuglydog.com -w /home/fweew92k6plb/public_html
[Tue Mar  3 15:11:53 MST 2020] Using stage ACME_DIRECTORY: https://acme-staging-v02.api.letsencrypt.org/directory
[Tue Mar  3 15:11:54 MST 2020] Multi domain='DNS:fuglydog.com,DNS:www.fuglydog.com'
[Tue Mar  3 15:11:54 MST 2020] Getting domain auth token for each domain
[Tue Mar  3 15:11:56 MST 2020] Getting webroot for domain='fuglydog.com'
[Tue Mar  3 15:11:56 MST 2020] Getting webroot for domain='www.fuglydog.com'
[Tue Mar  3 15:11:56 MST 2020] Verifying: fuglydog.com
[Tue Mar  3 15:11:59 MST 2020] fuglydog.com:Verify error:Invalid response from https://fuglydog.com/fuglydog/.well-known/acme-challenge/9tWbPvr7u90MRhbFsiFCjEeHgVH8Hr8OtvIECUi0mqA [107.180.48.175]:

acme.sh --test --issue -d fuglydog.com -d www.fuglydog.com -w /home/fweew92k6plb/public_html/fuglydog
[Tue Mar  3 15:10:16 MST 2020] Using stage ACME_DIRECTORY: https://acme-staging-v02.api.letsencrypt.org/directory
[Tue Mar  3 15:10:19 MST 2020] Multi domain='DNS:fuglydog.com,DNS:www.fuglydog.com'
[Tue Mar  3 15:10:19 MST 2020] Getting domain auth token for each domain
[Tue Mar  3 15:10:22 MST 2020] Getting webroot for domain='fuglydog.com'
[Tue Mar  3 15:10:23 MST 2020] Getting webroot for domain='www.fuglydog.com'
[Tue Mar  3 15:10:23 MST 2020] Verifying: fuglydog.com
[Tue Mar  3 15:10:28 MST 2020] Pending
[Tue Mar  3 15:10:30 MST 2020] Pending
[Tue Mar  3 15:10:33 MST 2020] Pending
[Tue Mar  3 15:10:37 MST 2020] fuglydog.com:Verify error:Invalid response from https://fuglydog.com/fuglydog/.well-known/acme-challenge/ZBcUR0JUtHVseKR-Ybo3KL9FO_BRvV0TKD1kAvamq5M [107.180.48.175]:

acme.sh --test --issue -d rozanosigns.com -d www.rozanosigns.com -w /home/fweew92k6plb/public_html
[Tue Mar  3 11:35:05 MST 2020] Using stage ACME_DIRECTORY: https://acme-staging-v02.api.letsencrypt.org/directory
[Tue Mar  3 11:35:05 MST 2020] Creating domain key
[Tue Mar  3 11:35:06 MST 2020] The domain key is here: /home/fweew92k6plb/.acme.sh/rozanosigns.com/rozanosigns.com.key
[Tue Mar  3 11:35:06 MST 2020] Multi domain='DNS:rozanosigns.com,DNS:www.rozanosigns.com'
[Tue Mar  3 11:35:06 MST 2020] Getting domain auth token for each domain
[Tue Mar  3 11:35:10 MST 2020] Getting webroot for domain='rozanosigns.com'
[Tue Mar  3 11:35:10 MST 2020] Getting webroot for domain='www.rozanosigns.com'
[Tue Mar  3 11:35:10 MST 2020] Verifying: rozanosigns.com
[Tue Mar  3 11:35:16 MST 2020] Pending
[Tue Mar  3 11:35:19 MST 2020] rozanosigns.com:Verify error:Invalid response from https://rozanosigns.com/rozanosigns/.well-known/acme-challenge/O5wTkRB7qQwjQxHdWrE8N2IU6_iC2UlyOq62UVfuITQ/ [107.180.48.175]:

**Comment by AndyZaa** [comment]
Neil,

I'm attaching a debug 2 log file for the failed cert issue. It seems like the script is looking for the verification challenge in the fuglydog directory rather than the root public_html directory. The script is actually creating an empty fuglydog.com directory in public_html.

[2020-03-03 FuglyDog Cert debug 2.txt](https://github.com/acmesh-official/acme.sh/files/4284023/2020-03-03.FuglyDog.Cert.debug.2.txt)

**Comment by AndyZaa** [comment]
Tonight I removed all directories and references to acme.sh and .well-known including all certs. I re-installed acme.sh and went through the installation process just as I did in 2018 using the exact steps and formats. Below are my results. I just can't figure out what could be going wrong. Is there a configuration somewhere that sets the path to the challenge file? Is there some parameter or config path that I can set to refer to the proper .well-known/acme-challenge path for each site?

acme.sh --issue -d zaayenga.net -d www.zaayenga.net -w /home/fweew92k6plb/public_html
 SUCCESS

acme.sh --issue -d biobanking.network -d www.biobanking.network -w /home/fweew92k6plb/public_html/biobanking
 SUCCESS

acme.sh --issue -d curl-upanddye.com -d www.curl-upanddye.com -w /home/fweew92k6plb/public_html/cuad
 SUCCESS

acme.sh --issue -d fuglydog.com -d www.fuglydog.com -w /home/fweew92k6plb/public_html/fuglydog
 ERROR WITH "Invalid response from https://fuglydog.com/fuglydog/.well-known/acme-challenge" (path with incorrect root)

acme.sh --issue -d rozanosigns.com -d www.rozanosigns.com -w /home/fweew92k6plb/public_html/rozanosigns 
 ERROR WITH "Invalid response from https://rozanosigns.com/rozanosigns/.well-known/acme-challenge" (path with incorrect root)

acme.sh --issue -d smarterlab.com -d www.smarterlab.com -w /home/fweew92k6plb/public_html/smarterlab 
 ERROR WITH "Invalid response from https://smarterlab.com/smarterlab/.well-known/acme-challenge" (path with incorrect root)

Thanks.

**Comment by AndyZaa** [comment]
Hi Neil,

I finally stumbled across my solution. You had mentioned to check my .htaccess files for changes and I confirmed they had not changed since before the problem started. However today I copied the .htaccess file from one of the successfully renewing sites to the failing sites. That did the trick! I don't know what could have crept into the .htaccess files except to think that a Wordpress plugin may have munged the entries and not changed the file date.

Due to me removing acme.sh and associated directories several times I did have to re-create the .well-known/acme-challenge directories in public_html and every addon domain directory with proper 755 permissions. Then I sailed through install & deploy on all the sites.

I deepfully apologize for any trouble I put you through for what obviously was an issue not of your making. I have closed the ticket. Thank you very much for your contributions in creating such a useful tool and putting up with noobies like me!

Andy
